### PR TITLE
Implement current source of ADC ADS114S08

### PR DESCRIPTION
**Pull request by benediktibk**
This extends the channel configuration to allow the selection of the output pins for the current sources within an ADC. It also implements this feature for the ADC ADS114S08, which requires the configuration of a pin mux within the ADC to be able to use the current sources.

### CODE REVIEW

**Comment by benediktibk** [comment]
depends on completion of #54911

File: include/zephyr/drivers/adc.h [hunk 1]
```diff
@@ -294,8 +308,8 @@ struct adc_dt_spec {
 	IF_ENABLED(DT_NODE_EXISTS(node_id), \
 		(.channel_cfg_dt_node_exists = true, \
 		 .channel_cfg  = ADC_CHANNEL_CFG_DT(node_id), \
-		 .vref_mv      = DT_PROP_OR(node_id, zephyr_vref_mv, 0), \
-		 .resolution   = DT_PROP_OR(node_id, zephyr_resolution, 0), \
+		 .vref_mv = DT_PROP_OR(node_id, zephyr_vref_mv, 0), \
+		 .resolution = DT_PROP_OR(node_id, zephyr_resolution, 0), \
```
Comment (MaureenHelm) [inline include/zephyr/drivers/adc.h:312]: Unrelated whitespace change

File: dts/bindings/adc/ti,ads114s08.yaml [hunk 2]
```diff
@@ -26,5 +26,21 @@ properties:
...
+      - 50
+      - 100
+      - 250
+      - 500
+      - 750
+      - 1000
+      - 1500
+      - 2000
+    default: 0
+    description: IDAC current in microampere
```
Comment (MaureenHelm) [inline dts/bindings/adc/ti,ads114s08.yaml:43]: The default value needs a justification in the description

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: include/zephyr/drivers/adc.h (hunk 1 above)
Comment (benediktibk) [reply include/zephyr/drivers/adc.h]: done

File: dts/bindings/adc/ti,ads114s08.yaml (hunk 2 above)
Comment (benediktibk) [reply dts/bindings/adc/ti,ads114s08.yaml]: done (hopefully sufficiently detailed)

**Review by MaureenHelm — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]
